src/Microsoft.AspNetCore.Server.Kestrel/Http/FrameOfT.cs

```diff
@@ -37,8 +37,14 @@ public class Frame<TContext> : Frame
                     {
                         if (SocketInput.RemoteIntakeFin)
                         {
+                            if (TakeStartLine(SocketInput))
```

Comment:
The pattern is odd since you have a guard for ` && !TakeStartLine(SocketInput)` so does this scan again if RemoteIntakeFin is true?

---
In reply to: [56759425](https://github.com/aspnet/KestrelHttpServer/pull/705#discussion_r56759425) [](ancestors = 56759425)